packages/modular-scripts/src/utils/options.ts

```diff
@@ -0,0 +1,45 @@
+import os from 'os';
+
+/**
+ * Validates comparison options in combination.
+ *
+ * Throws and exits when invalid combinations are detected.
+ */
+export function validateCompareOptions(
+  compareBranch: string | undefined,
+  changed: boolean,
+): void {
+  if (compareBranch && !changed) {
+    process.stderr.write(
+      "Option --compareBranch doesn't make sense without option --changed\n",
+    );
+    process.exit(1);
+  }
+}
+
+/**
+ * Validates and converts concurrency option to number.
+ *
+ * Throws and exits when invalid value is detected.
+ */
+export function computeConcurrencyOption(
+  concurrencyLevel: string | undefined,
+): number {
+  if (!concurrencyLevel) {
+    // If concurrencyLevel is not supplied, default to the number of CPUs. If the number of CPUs is 0 (/proc or equivalent is unavailable), default to no concurrency.
+    return os.cpus().length || 1;
+  }
+
+  // Otherwise, try to parse it
+  const concurrency = parseInt(concurrencyLevel, 10);
+
+  // If not a number or a negative number, bail out with an error.
+  if (isNaN(concurrency) || concurrency < 0) {
```

Comment:
What happens if someone provides a big number? Let's say they typo a couple zeros on the end and put `600`. Is this fine or do we need an upper boundary?

Reply:
> What happens if someone provides a big number? Let's say they typo a couple zeros on the end and put `600`. Is this fine or do we need an upper boundary?

It's fine, because:
1) in practice, the number of things a project can build concurrently at a given stage will be much less than 600 (even if the project has 600 packages, they will be probably interdependent)
2) the worse that can happen is that Modular goes out of memory, in which case the user will just correct their command